docs/self-hosting/configuration/domain-configuration.mdx

```diff
@@ -27,36 +27,165 @@ NEXTAUTH_URL=https://formbricks.example.com
+
+#### Multi-Domain Setup SDK Configuration
+
+When using PUBLIC_URL (multi-domain setup), all SDK configurations must use the **public domain**:
+
+```js
+// JavaScript/React/Vue.js/Next.js SDKs
+formbricks.setup({
+  environmentId: "your-environment-id",
+  appUrl: "https://surveys.example.com", // Use PUBLIC_URL domain
+});
+```
 
-- The `SURVEY_URL` must be a valid URL with protocol
-- When `SURVEY_URL` is set, all public survey links will use this domain.
-- If `SURVEY_URL` is not set, surveys will be served from the `WEBAPP_URL` domain
-- Make sure your DNS and SSL certificates are properly configured for both domains
-- The survey domain should be properly configured in your reverse proxy or web server
-- **Important**: The second domain is exclusively for serving surveys. Any non-survey paths accessed on this domain will return a 404 error. This is a security feature to ensure the survey domain only serves survey content.
+```jsx
+// React Native SDK
+await formbricks.setup({
+  environmentId: "your-environment-id",
+  appUrl: "https://surveys.example.com", // Use PUBLIC_URL domain
+});
+```
+
+```swift
+// iOS Swift SDK
+FormbricksSDK.shared.configure(
+  appUrl: "https://surveys.example.com", // Use PUBLIC_URL domain
+  environmentId: "yourEnvironmentId"
+)
+```
+
+#### Single Domain Setup SDK Configuration
+
+When PUBLIC_URL is not configured (single domain setup), SDKs use the admin domain:
+
+```js
+formbricks.setup({
+  environmentId: "your-environment-id",
+  appUrl: "https://formbricks.example.com", // Use WEBAPP_URL domain
+});
+```
+
+#### Key SDK Changes Made
+
+- **Component Updates**: `ConnectWithFormbricks` component now uses `publicDomain` prop instead of `webAppUrl`
+- **Setup Instructions**: All setup instructions dynamically show the correct domain based on your configuration
+- **Connection Flow**: Environment connection flow automatically uses the appropriate domain
+- **File Uploads**: Public file uploads now route through the public domain
+- **Survey Rendering**: Survey widgets and embeds automatically connect to the public domain
+
+#### Migration for Existing SDK Implementations
+
+If you're upgrading from a previous version:
+
+1. **Check your current setup**: Identify if you're using single or multi-domain configuration
+2. **Update appUrl parameter**: Ensure it points to the correct domain based on your setup
+3. **Test connections**: Verify that surveys load correctly after the change
+4. **Update documentation**: Update any internal documentation referencing the old domain structure
+
+### DNS and SSL Configuration
+
+1. **Set up DNS records** for both domains
+2. **Configure SSL certificates** for both domains
+3. **Update your reverse proxy** (nginx, Apache, Cloudflare, etc.) to route both domains to your Formbricks instance
+4. **Firewall rules**: Ensure both domains can reach your Formbricks instance
+
+### Important Security Considerations
```

Comment:
I would suggest that we use this information in the introduction of the feature and avoid having this multiple times on the same documentation page.